source/synthDrivers/sapi4.py

```diff
@@ -94,6 +115,349 @@ def IUnknown_Release(self, this, *args, **kwargs):
+		if not self._waveFormat:
+			raise ReturnHRESULT(AUDERR_NEEDWAVEFORMAT, None)
+		with self._audioCond:
+			if self._deviceUnClaiming:
+				# Unclaiming is cancelled, but nothing else is touched.
+				self._deviceUnClaiming = False
+				self._deviceUnClaimingBytePos = None
+				return
+		if self._deviceClaimed:
+			raise ReturnHRESULT(AUDERR_ALREADYCLAIMED, None)
+		self._maybeInitPlayer()
+		self._deviceClaimed = True
+		if self._notifySink:
+			self._notifySink.AudioStart()
+
+	def IAudio_UnClaim(self) -> None:
+		"""Releases the multimedia device asynchronously.
+		Called after the engine completes writing all audio data.
+		If there is audio in the buffer, it should still be played till the end.
+		`IAudioDestNotifySink::AudioStop()` will be called after the audio completely stops."""
+		if not self._deviceClaimed:
+			raise ReturnHRESULT(AUDERR_NOTCLAIMED, None)
+		if self._deviceStarted:
+			# When playing, wait for the playback to finish.
+			with self._audioCond:
+				self._deviceUnClaiming = True
+				self._deviceUnClaimingBytePos = self._writtenBytes
+				self._audioCond.notify()
+		else:
+			# When not playing, this can finish immediately.
+			if self._writtenBytes == self._playedBytes and not self._audioQueue:
+				# If all audio is done playing, stop the player.
+				self._player.stop()
+			self._deviceClaimed = False
+			if self._notifySink:
+				self._notifySink.AudioStop(0)  # IANSRSN_NODATA
+
+	def IAudio_Start(self) -> None:
+		"""Starts (or resumes) playing the audio in the buffer."""
+		if self._deviceStarted:
+			raise ReturnHRESULT(AUDERR_ALREADYSTARTED, None)
+		if not self._deviceClaimed:
+			raise ReturnHRESULT(AUDERR_NOTCLAIMED, None)
+		self._startTime = datetime.now()
+		self._startBytes = self._playedBytes
+		try:
+			self._player.pause(False)
+		except OSError:
+			log.debugWarning("Error starting audio", exc_info=True)
+		with self._audioCond:
+			self._deviceStarted = True
+			self._audioCond.notify()
+
+	def IAudio_Stop(self) -> None:
+		"""Stops (or pauses) playing, without clearing the buffer.
+		If there is audio in the buffer, calling Stop and UnClaim should keep the buffer
+		and only pause the playback."""
+		if not self._deviceStarted:
+			return  # no error returned
+		try:
+			self._player.pause(True)
+		except OSError:
+			log.debugWarning("Error stopping audio", exc_info=True)
+		with self._audioCond:
+			self._deviceStarted = False
+			self._audioCond.notify()
+
+	def IAudio_TotalGet(self) -> int:
+		"""Returns the total number of bytes written,
+		including the unplayed bytes in the buffer,
+		which should increase monotonically and never reset."""
+		return self._writtenBytes
+
+	def IAudio_ToFileTime(self, pqWord: c_ulonglong_p) -> None:
+		"""Converts a byte position to UTC FILETIME."""
+		if not self._waveFormat:
+			raise ReturnHRESULT(AUDERR_NEEDWAVEFORMAT, None)
+		filetime_ticks = int((self._startTime.timestamp() + 11644473600) * 10_000_000)
```

Comment:
can you turn this magic number into a named constant?

```suggestion
		UNIX_TIME_CONV = 1_1644_473_600
		filetime_ticks = int((self._startTime.timestamp() + UNIX_TIME_CONV) * 10_000_000)
```